pkg/smartcontract/param_type.go

```diff
@@ -184,7 +188,7 @@ func ParseParamType(typ string) (ParamType, error) {
 }
 
 // adjustValToType is a value type-checker and converter.
-func adjustValToType(typ ParamType, val string) (interface{}, error) {
+func adjustValToType(typ ParamType, typStr, val string) (interface{}, error) {
```

Comment:
I think we're better not change the interface here, it can be handled in the `NewParameterFromString`.